1: Loading dataset

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

import warnings
warnings.filterwarnings('ignore')

In [5]:
train_trans = pd.read_csv('train_transaction.csv')
train_id = pd.read_csv('train_identity.csv')

In [6]:
train_trans.shape, train_id.shape

((590540, 394), (144233, 41))

In [7]:
train_trans.head()

TransactionID  isFraud  TransactionDT  TransactionAmt ProductCD  card1  \
0        2987000        0          86400            68.5         W  13926   
1        2987001        0          86401            29.0         W   2755   
2        2987002        0          86469            59.0         W   4663   
3        2987003        0          86499            50.0         W  18132   
4        2987004        0          86506            50.0         H   4497   

   card2  card3       card4  card5  ... V330  V331  V332  V333  V334 V335  \
0    NaN  150.0    discover  142.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   
1  404.0  150.0  mastercard  102.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   
2  490.0  150.0        visa  166.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   
3  567.0  150.0  mastercard  117.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   
4  514.0  150.0  mastercard  102.0  ...  0.0   0.0   0.0   0.0   0.0  0.0   

  V336  V337  V338  V339  
0  NaN   NaN   NaN   NaN  
1  NaN   NaN   NaN   NaN  
2  NaN   NaN   NaN   NaN  
3  NaN   NaN   NaN   NaN  
4  0.0   0.0   0.0   0.0  

[5 rows x 394 columns]

In [8]:
train_id.head()

TransactionID  id_01     id_02  id_03  id_04  id_05  id_06  id_07  id_08  \
0        2987004    0.0   70787.0    NaN    NaN    NaN    NaN    NaN    NaN   
1        2987008   -5.0   98945.0    NaN    NaN    0.0   -5.0    NaN    NaN   
2        2987010   -5.0  191631.0    0.0    0.0    0.0    0.0    NaN    NaN   
3        2987011   -5.0  221832.0    NaN    NaN    0.0   -6.0    NaN    NaN   
4        2987016    0.0    7460.0    0.0    0.0    1.0    0.0    NaN    NaN   

   id_09  ...                id_31  id_32      id_33           id_34  id_35  \
0    NaN  ...  samsung browser 6.2   32.0  2220x1080  match_status:2      T   
1    NaN  ...   mobile safari 11.0   32.0   1334x750  match_status:1      T   
2    0.0  ...          chrome 62.0    NaN        NaN             NaN      F   
3    NaN  ...          chrome 62.0    NaN        NaN             NaN      F   
4    0.0  ...          chrome 62.0   24.0   1280x800  match_status:2      T   

  id_36 id_37  id_38  DeviceType                     DeviceInfo  
0     F     T      T      mobile  SAMSUNG SM-G892A Build/NRD90M  
1     F     F      T      mobile                     iOS Device  
2     F     T      T     desktop                        Windows  
3     F     T      T     desktop                            NaN  
4     F     T      T     desktop                          MacOS  

[5 rows x 41 columns]

2:Check the data memory:

In [9]:
memory_trans = train_trans.memory_usage().sum() / 1024**2
print('Memory usage of dataframe is {:.2f} MB'.format(memory_trans))

Memory usage of dataframe is 1775.15 MB


In [10]:
memory_id = train_id.memory_usage().sum() / 1024**2
print('Memory usage of dataframe is {:.2f} MB'.format(memory_id))

Memory usage of dataframe is 45.12 MB


Write function for check the columns datatype

In [12]:
np.finfo(np.float64).max,np.finfo(np.float64).min,np.finfo(np.float32).max,np.finfo(np.float32).min,np.finfo(np.float16).max,np.finfo(np.float16).min

(1.7976931348623157e+308,
 -1.7976931348623157e+308,
 3.4028235e+38,
 -3.4028235e+38,
 65500.0,
 -65500.0)

In [72]:
np.iinfo(np.int64).max,np.iinfo(np.int64).min,np.iinfo(np.int32).max,np.iinfo(np.int32).min,np.iinfo(np.int16).max,np.iinfo(np.int16).min

(9223372036854775807,
 -9223372036854775808,
 2147483647,
 -2147483648,
 32767,
 -32768)

Conver the type and decrease our memory 

In [13]:
f64 = train_trans.select_dtypes(include ='float64')
for i in f64:
    if train_trans[i].max() < 65500 and train_trans[i].min() > -65500:
        train_trans[i] = train_trans[i].astype(np.float16)
    elif train_trans[i].max() < 3.4028235e+38 and train_trans[i].min() > -3.4028235e+38:
        train_trans[i] = train_trans[i].astype(np.float32)
    else:
        train_trans[i] = train_trans[i].astype(np.float64)
        
    

In [74]:
int64 = train_trans.select_dtypes(include ='int64')
for i in int64:
    if train_trans[i].max() < 32767 and train_trans[i].min() > -32768:
        train_trans[i] = train_trans[i].astype(np.int16)
    elif train_trans[i].max() < 2147483647 and train_trans[i].min() > -2147483648:
        train_trans[i] = train_trans[i].astype(np.int32)
    else:
        train_trans[i] = train_trans[i].astype(np.int64)
        
        


        TransactionID  TransactionDT
0             2987000          86400
1             2987001          86401
2             2987002          86469
3             2987003          86499
4             2987004          86506
...               ...            ...
590535        3577535       15811047
590536        3577536       15811049
590537        3577537       15811079
590538        3577538       15811088
590539        3577539       15811131

[590540 rows x 2 columns]


In [14]:
memory_trans = train_trans.memory_usage().sum() / 1024**2
print('Memory usage of dataframe is {:.2f} MB'.format(memory_trans))

Memory usage of dataframe is 554.17 MB


for the train_trans file the memory become 642.03 MB after convert. for the train_id file 45.12 MB is small enough to operate.
